### PR TITLE
Merge from vuejs-templates/webpack#develop preserving PWA configuration

### PR DESCRIPTION
**Pull request by NicoCevallos**
Big merge from `vuejs-templates/webpack#develop`
Main changes:
- Switch to `webpack-dev-server`. Same intention as the PR [118](https://github.com/vuejs-templates/pwa/pull/118) that applies the webpack template's PR [975](https://github.com/vuejs-templates/webpack/pull/975). It adds new dependencies and remove unused dependencies
  - Removed
    - `connect-history-api-fallback` used in `template/build/dev-server.js`
    - `eventsource-polyfill` used in `template/build/dev-client.js`
    - `express` used in `template/build/dev-server.js`
    - `http-proxy-middleware` used in `template/build/dev-server.js`
    - `opn` used in `template/build/dev-server.js`
    - `webpack-dev-middleware` used in `template/build/dev-server.js`
    - `webpack-hot-middleware` used in `template/build/dev-client.js` and `template/build/dev-server.js`
  - Added
    - `friendly-errors-webpack-plugin`
    - `html-webpack-plugin`
    - `node-notifier`
    - `portfinder`
    - `webpack-bundle-analyzer`
    - `webpack-dev-server`
- Merge documentation fixing some typos but preserving the customization to PWA
- Merge template initialization adding options to use `jest` and `yarn`
- Remove fixes to add `,` or `;` for airbnb ESLint preset, due webpack template runs `lint --fix` if the user select to intall dependencies using `npm` or `yarn`
- Merge `package.json` to:
  - add `eslint-plugin-vue` to lint HTML Templates
  - Add `postcss` modules
  - Remove `cssnano` it is not present on webpack template, but as it's a dependency of `optimize-css-assets-webpack-plugin` and `css-loader`, so it is not required on the package.json
- Sort the dependencies alphabetically to ease next merges/pull requests

Thanks in advance for any review and comment.

### CODE REVIEW

**Comment by anubhav7495** [comment]
@nicocevallos Thanks for the contribution. Can you please break this into multiple PR's according to the way they were merged in Webpack template. Currently it is too big to actually review.

Thanks.

**Comment by NicoCevallos** [comment]
@anubhav7495 I simply compared both repos and move the changes, preserving anything related to PWA in `webpack.base.conf.js` and documentation, due to it looks like a lot of merged PR to Webpack template was not merged to this one.
Do you have a way to check the PRs merged to Webpack template that needs to be merged with this one?
Another option is to do the same but, progressively.

Let me know what do you think.

**Comment by michalsnik** [comment]
@anubhav7495 any news here? It would be really nice to have jest setup in this template too

**Comment by cristianl** [comment]
@michalsnik As the next version of vue-cli moves to a plugin-based architecture instead of templates, I don't think the PWA template will get new features. With the 3.0 beta [coming very soon](https://github.com/vuejs/vue-cli/issues/589#issuecomment-362972980), the final release is unlikely to take long.

**Comment by michalsnik** [comment]
I know @cristianl, but it will take some time for the new cli to be production ready. And this template already is, so it would be faster to merge those changes anyway.

**Comment by cristianl** [comment]
@michalsnik Oh, I see – sorry for the 'splaining.

File: docs/backend.md [hunk 1]
```diff
@@ -5,18 +5,37 @@ If you are building a purely-static app (one that is deployed separately from th
...
 
 ``` js
+// config/index.js
 'use strict'
-
 const path = require('path')
 
 module.exports = {
+  dev: {
+    / Paths
```
Comment (niutech) [inline docs/backend.md:14]: Missing `//`

**Comment by NicoCevallos** [comment]
Hello there!

I should close this PR, right? I guess nobody will find those changes useful with the vue-cli 3.
What do you think @anubhav7495?

**Pull request closed without merging**